### PR TITLE
Add ability to retrieve taglines from site

### PR DESCRIPTION
**Pull request by micahmo**
This PR adds the ability to retrieve [Taglines](https://join-lemmy.org/api/interfaces/Tagline.html) from a Lemmy site. The goal is to show the tagline at the top of the Thunder UI (like Jerboa).

Here is an example of a tagline on [programming.dev](https://programming.dev).

| image |
| - |

Here's how we can retrieve it via the API now.

``` dart
FullSiteView fullSiteView = await lemmy.run(
  GetSite(
    auth: response.jwt!.raw,
  ),
);

print(fullSiteView.taglines?.first.content);
```

And here's what the output would look like.

```
If you recently signed up (since Jul 9) and used an email, it most likely did not get saved properly. You will need to add it in your profile settings again.
```

### CODE REVIEW

File: lib/src/v3/models/api.dart [hunk 1]
```diff
@@ -123,6 +124,7 @@ class FullSiteView with _$FullSiteView {
     MyUserInfo? myUser,
     FederatedInstances? federatedInstances,
     required String instanceHost,
+    List<Tagline>? taglines,
```
Comment (hjiangsu) [inline lib/src/v3/models/api.dart:127]: This should be a required field, and if theres no taglines, it should be an empty list `[]`!

This follows the lemmy api: https://join-lemmy.org/api/interfaces/GetSiteResponse.html#taglines

File: lib/src/v3/models/api.dart [hunk 2]
```diff
@@ -131,6 +133,23 @@ class FullSiteView with _$FullSiteView {
       _$FullSiteViewFromJson(json);
 }
 
+@freezed
+class Tagline with _$Tagline {
+  @modelSerde
+  const factory Tagline({
+    int? id,
```
Comment (hjiangsu) [inline lib/src/v3/models/api.dart:140]: Everything here should be required fields as well to match up with the lemmy api: https://join-lemmy.org/api/interfaces/Tagline.html

I think the only thing missing here is the `updated` param which should be optional!

**Review by hjiangsu — CHANGES_REQUESTED** [review]
Just have a few comments!

File: lib/src/v3/models/api.dart (hunk 1 above)
Comment (micahmo) [reply lib/src/v3/models/api.dart]: Done.

File: lib/src/v3/models/api.dart (hunk 2 above)
Comment (micahmo) [reply lib/src/v3/models/api.dart]: Done.

**Comment by micahmo** [comment]
@hjiangsu Will you handle updating this dependency in Thunder so that the new field can be used?

**Comment by hjiangsu** [comment]
Yup, I'll handle it!

**Pull request merged**